src/team_lib/hier/xccl_hier_context.c

```diff
@@ -34,24 +33,24 @@ xccl_hier_init_tl(xccl_hier_context_t *ctx, xccl_tl_id_t tl_id,
         .thread_mode     = XCCL_THREAD_MODE_SINGLE,
         .completion_type = XCCL_TEAM_COMPLETION_TYPE_BLOCKING,
         .oob             = oob,
-        .tls             = 1 << tl_id,
+        .tls             = tl_id,
     };
 
-    if (XCCL_OK != xccl_lib_init(&lib_params, NULL, &lib)) {
-        return XCCL_ERR_NO_MESSAGE;
-    }
-
-    if (prefix != NULL) {
+    if (hier_cfg->super.env_prefix != NULL) {
         snprintf(env_prefix, sizeof(env_prefix), "%s_HIER_%s",
-                 prefix, xccl_tl_str(1 << tl_id));
+                 hier_cfg->super.env_prefix, xccl_tl_str(tl_id));
     }
     else {
-        snprintf(env_prefix, sizeof(env_prefix), "HIER_%s", xccl_tl_str(1 << tl_id));
+        snprintf(env_prefix, sizeof(env_prefix), "HIER_%s", xccl_tl_str(tl_id));
     }
-
     xccl_context_config_read(lib, env_prefix, NULL, &cfg);
+    if (hier_cfg->devices.count > 1 ||
```

Comment:
Maybe we should allow to pass 'all' as net_devices? UCX recognizes this as "all available devices"

Reply:
The idea is:  if user does not specify XCCL_HIER_NET_DEVICES then we should not overwrite potential HIER_ucx_XCCL_TEAM_UCX_NET_DEVICES with "all".